src/dma.rs

```diff
@@ -59,7 +59,9 @@ impl<B, C: Channel, T: Target> Transfer<B, C, T> {
         let (ptr, len) = unsafe { buffer.write_buffer() };
         let len = u16(len).expect("buffer is too large");
 
-        channel.set_memory_address(ptr as u32, Increment::Enable);
+        // NOTE(unsafe) We are using the address of a 'static WriteBuffer here,
+        // which is guaranteed to be safe for DMA.
+        unsafe { channel.set_memory_address(ptr as u32, Increment::Enable) };
```

Comment:
I agree, but they sort of have to be here, because otherwise the code wouldn't compile with the current nightly version of the PACs. What happened is that the PACs have made writing the memory address registers of the DMA `unsafe`, which forces us to adapt.

I could pull out these changes into a separate "update PACs" PR so we can merge it separately. But this PR would still need to be based off that other PR, so you would still see those changes here, so reviewing wouldn't get easier.

Reply:
Thanks for clearing that up.